pkg/controllers/updaterun/execution_integration_test.go

```diff
@@ -972,6 +970,349 @@ var _ = Describe("UpdateRun execution tests", func() {
+			binding := resourceBindings[2] // cluster-3
+			validateBindingState(ctx, binding, resourceSnapshot.Name, updateRun, 0)
+
+			By("Updating the 3rd clusterResourceBinding to Available")
+			meta.SetStatusCondition(&binding.Status.Conditions, generateTrueCondition(binding, placementv1beta1.ResourceBindingAvailable))
+			Expect(k8sClient.Status().Update(ctx, binding)).Should(Succeed(), "failed to update the binding status")
+
+			By("Validating the 3rd cluster has succeeded and 4th cluster has started")
+			wantStatus.StagesStatus[0].Clusters[2].Conditions = append(wantStatus.StagesStatus[0].Clusters[2].Conditions, generateTrueCondition(updateRun, placementv1beta1.ClusterUpdatingConditionSucceeded))
+			wantStatus.StagesStatus[0].Conditions[0] = generateFalseCondition(updateRun, placementv1beta1.StageUpdatingConditionProgressing) // The progressing condition now becomes false with waiting reason.
+			wantStatus.StagesStatus[0].AfterStageTaskStatus[0].Conditions = append(wantStatus.StagesStatus[0].AfterStageTaskStatus[0].Conditions,
+				generateTrueCondition(updateRun, placementv1beta1.AfterStageTaskConditionApprovalRequestCreated))
+			wantStatus.Conditions[1] = generateFalseCondition(updateRun, placementv1beta1.StagedUpdateRunConditionProgressing)
+			validateClusterStagedUpdateRunStatus(ctx, updateRun, wantStatus, "")
+		})
+
+		It("Should complete the 1st stage after wait time passed and approval request approved", func() {
+			By("Validating the approvalRequest has been created")
+			approvalRequest := &placementv1beta1.ClusterApprovalRequest{}
+			wantApprovalRequest := &placementv1beta1.ClusterApprovalRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: updateRun.Status.StagesStatus[0].AfterStageTaskStatus[0].ApprovalRequestName,
+					Labels: map[string]string{
+						placementv1beta1.TargetUpdatingStageNameLabel:   updateRun.Status.StagesStatus[0].StageName,
+						placementv1beta1.TargetUpdateRunLabel:           updateRun.Name,
+						placementv1beta1.IsLatestUpdateRunApprovalLabel: "true",
+					},
+				},
+				Spec: placementv1beta1.ApprovalRequestSpec{
+					TargetUpdateRun: updateRun.Name,
+					TargetStage:     updateRun.Status.StagesStatus[0].StageName,
+				},
+			}
+			Eventually(func() error {
+				if err := k8sClient.Get(ctx, types.NamespacedName{Name: wantApprovalRequest.Name}, approvalRequest); err != nil {
+					return err
+				}
+				if diff := cmp.Diff(wantApprovalRequest.Spec, approvalRequest.Spec); diff != "" {
+					return fmt.Errorf("approvalRequest has different spec (-want +got):\n%s", diff)
+				}
+				if diff := cmp.Diff(wantApprovalRequest.Labels, approvalRequest.Labels); diff != "" {
+					return fmt.Errorf("approvalRequest has different labels (-want +got):\n%s", diff)
+				}
+				return nil
+			}, timeout, interval).Should(Succeed(), "failed to validate the approvalRequest")
+
+			By("Approving the approvalRequest")
+			approveClusterApprovalRequest(ctx, wantApprovalRequest.Name)
```

Comment:
Before approving the request, try first delete it and verify it's recreated instantly?